### ISSUE TITLE
Normalized Torque Curve for Gas Motor

### ISSUE THREAD

**Issue by RandomationGames** [RandomationGames/Randomation-Vehicle-Physics#7]
Currently, the torque curve on the GasMotor script represents both the RPM range and the torque of the engine across the range. This means that the top speed of a vehicle is directly impacted by the width of the curve. The problem with this is that it's difficult to change the top speed of a vehicle while maintaining the same torque levels across the RPM range.

I have contemplated adding an option for the torque curve to be normalized with regard to its RPM range. With this, a separate variable could be used to set the maximum RPM of the engine while the torque curve remains unchanged. The torque curve will be sampled at a normalized point based on the current output RPM of the engine relative to its maximum RPM.

Basically, the x-values of the torque curve will range from 0 to 1, and the sampled value in this range will be the fraction of the engine's current RPM out of the maximum RPM.
